Conflict with Sphinx 5

**Describe the bug**
Sphinx 5, [released end of May](https://www.sphinx-doc.org/en/master/changes.html), doesn't work, while the previous version does.

**To Reproduce**
Follow the contributers' guide for setting up. Running the tests fails. To reduce this further, just run the "doc" session:
```
python3 -m nox --nocolor --session doc
nox > Running session doc
nox > Creating virtual environment (virtualenv) using python3 in .nox/doc
nox > python -m pip install -r doc/requirements.txt docutils
nox > python -m pip install -e .
nox > cd doc
nox > make html O=-W
gcovr -h > examples/gcovr.out
Running Sphinx v5.0.1

Warning, treated as error:
Invalid configuration value found: 'language = None'. Update your configuration to a valid langauge code. Falling back to 'en' (English).
make: *** [Makefile:27: html] Error 2
nox > Command make html O=-W failed with exit code 2
nox > Session doc failed.
```

**Expected behavior**

If you just replace the line "sphinx" in doc/requirements.txt with "sphinx < 5", you get this instead:
```
python3 -m nox --nocolor --session doc
nox > Running session doc
nox > Creating virtual environment (virtualenv) using python3 in .nox/doc
nox > python -m pip install -r doc/requirements.txt docutils
nox > python -m pip install -e .
nox > cd doc
nox > make html O=-W
gcovr -h > examples/gcovr.out
Running Sphinx v4.5.0
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 25 source files that are out of date
updating environment: 0 added, 6 changed, 0 removed
reading sources... [100%] license                                                                                                                                                                           
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] output/txt                                                                                                                                                                         
generating indices... genindex done
writing additional pages... search done
copying images... [100%] ../images/screenshot-html-details.example.cpp.png                                                                                                                                  
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in build/html.
nox > cd ..
nox > rst2html5.py --strict README.rst .nox/doc/tmp/README.html
nox > Session doc was successful.
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 21.10
 - GCC version: multiple (8,9,11)
 - GCOVR version: 5.1, master

--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -79,7 +79,7 @@ master_doc = "index"
 #
 # This is also used if you do content translation via gettext catalogs.
 # Usually you set "language" from the command line for these cases.
-language = None
+language = "en"
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.

--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -191,4 +191,4 @@ todo_include_todos = True
 # -- Options for extlinks extension ------------------------------------------
 
 # see http://www.sphinx-doc.org/en/master/ext/extlinks.html
-extlinks = {"issue": ("https://github.com/gcovr/gcovr/issues/%s", "#")}
+extlinks = {"issue": ("https://github.com/gcovr/gcovr/issues/%s", "#%s")}

